### ISSUE TITLE
Different result between pytorch exported .onnx & .nn

### ISSUE THREAD

**Issue by dytan** [Unity-Technologies/barracuda-release#56]
Hi, something strange found when I try to run inference of a simple regression model trained with pytorch (several dense layers).
 
If directly use the onnx file from torch.onnx.export(...), the output is not right with pytorch. But after load the onnx with keras then converted to .nn with script 'Toos\keras_to_barracuda.py', the output is ok.

The inference is called with some code like this:
``` c#

public NNModel modelSource;

private Dictionary<string, Tensor> _inputs;
private IWorker _worker;
private Model _model;
private Tensor _w_new, _s;
 

void Start(){
    _model = ModelLoader.Load(modelSource);
    _worker = WorkerFactory.CreateWorker(_model, WorkerFactory.Device.CPU, true);

    _w_new = new Tensor(1, 9);
    _s = new Tensor(1,  256);

    _inputs = new Dictionary<string, Tensor>();
    _inputs.Add("w_new", _w_new);
    _inputs.Add("s", _s);
}


void Update(){
	int nObjSpace = 128;
	for (int i = 0; i < nObjSpace; i++){
        _s[i] = 0.2f;
		_s[i + nObjSpace] = 0.5f;
    }

    float[] w_new = {-0.005846714f, 0.01887251f, 0.1489416f, 0f,0f,0f,0f,0f,0f};
    for (int i=0; i<9; i++){
    	_w_new[i] = w_new[i];
    }

    _inputs["s"] = _s.DeepCopy();
    _inputs["w_new"] = _w_new.DeepCopy();

    _worker.Execute(_inputs);

    var r_new = _worker.PeekOutput("z_new");

}
```
Noticed torch exported .onnx treat dense as 'GeMM' while .nn treat as 'Dense' with kernel and bias, but the weights stuff is definitely same except the shape or layout. After doing some debug, found the two models already have different out after the first dense layer, and the question is they all come to this function to eval 'Dense' with the same args(no obvious difference found in the 'X' 'W' 'B' ). Besides, if doing inference with Microsoft.ML.OnnxRuntime on the .onnx, result is right too.

![Image](https://user-images.githubusercontent.com/11719101/83962037-0cd06480-a8cc-11ea-8357-21b13dcd9b95.png)

the example output is like this:
- torch .onnx
![Image](https://user-images.githubusercontent.com/11719101/83962195-fe834800-a8cd-11ea-9e30-a373116454d5.png)
- torch .onnx -> keras .h5 -> barracuda .nn (consistent with pytorch)
![Image](https://user-images.githubusercontent.com/11719101/83962206-1f4b9d80-a8ce-11ea-8963-1b632d22df5c.png)

Tested with barracuda version 0.7.1 & 1.0.0, and there are the models:
[models.zip](https://github.com/Unity-Technologies/barracuda-release/files/4741487/models.zip)

**Comment by AlexRibard** [comment]
Hi @dytan !
You are correct, there is indeed a discrepancy when executing the model.
The problem is on our end on the first `Concat` layer.
On 1.0, we have a bug when concatenating two 2 dim inputs.
I have pushed a fix, but in the meantime can you try to concat the two inputs manualy before and not in the model.

**Comment by dytan** [comment]
Ok, got it. Thank you!
